src/unity/python/turicreate/toolkits/object_detector/_sframe_loader.py

```diff
@@ -154,8 +294,28 @@ def __init__(self,
                     "the latter being a dictionary that defines 'x', 'y', 'width', and 'height'.\n"
                     "The following row(s) did not conform to this format:" + s)
 
-        # Convert images to raw to eliminate overhead of decoding
-        self.sframe[_TMP_COL_RAW_IMAGE] = self.sframe[self.feature_column].apply(_convert_image_to_raw)
+        # Compute the number of times we'll read a row from the SFrame.
+        sample_limits = []
+        if iterations is not None:
+            sample_limits.append(iterations * batch_size)
+        if epochs is not None:
+            sample_limits.append(epochs * len(sframe))
+        samples = min(sample_limits) if len(sample_limits) > 0 else None
+        if use_io_threads:
+            # Delegate SFrame operations to a background thread, leaving this
+            # thread to Python-based work of copying to MxNet and scheduling
+            # augmentation work in the MXNet backend.
+            #
+            # Note that the large buffer size is an attempt to mitigate against
+            # the SFrame shuffle operation that can occur after each epoch.
+            self.data_source = _SFrameAsyncDataSource(
+                sframe, feature_column, annotations_column,
+                load_labels=load_labels, shuffle=shuffle, samples=samples,
+                buffer_size=batch_size*8)
```

Comment:
Yeah, 8 was just the largest number I felt comfortable putting in, given the default batch size and the image sizes. Honestly I'm not-so-secretly hoping that we can improve the SFrame-sorting situation enough that we can largely do away with this parameter, since the main motivation for having up to 8 batches in flight is to buffer against that sort operation.

So it's a bit weird, on iMac Pro the GPU is pretty fast, so it's unlikely the SFrame stuff can keep up anyway. On MBP, this has more impact, at present.

Yeah, let me go ahead and put this into _advanced_parameters for now